source/common/config/filter_json.cc

```diff
@@ -282,5 +289,42 @@ void FilterJson::translateBufferFilter(const Json::Object& json_buffer,
   JSON_UTIL_SET_DURATION_SECONDS(json_buffer, buffer, max_request_time);
 }
 
+void FilterJson::translateTcpProxy(const Json::Object& json_tcp_proxy,
+                                   envoy::api::v2::filter::network::TcpProxy& tcp_proxy) {
+  json_tcp_proxy.validateSchema(Json::Schema::TCP_PROXY_NETWORK_FILTER_SCHEMA);
+
+  JSON_UTIL_SET_STRING(json_tcp_proxy, tcp_proxy, stat_prefix);
+  translateRepeatedAccessLog(json_tcp_proxy.getObjectArray("access_log", true),
+                             *tcp_proxy.mutable_access_log());
+
+  for (const Json::ObjectSharedPtr& route_desc :
+       json_tcp_proxy.getObject("route_config")->getObjectArray("routes")) {
+    envoy::api::v2::filter::network::TcpProxy::DeprecatedV1::TCPRoute* route =
+        tcp_proxy.mutable_deprecated_v1()->mutable_routes()->Add();
+    JSON_UTIL_SET_STRING(*route_desc, *route, cluster);
+    JSON_UTIL_SET_STRING(*route_desc, *route, destination_ports);
+    JSON_UTIL_SET_STRING(*route_desc, *route, source_ports);
+
+    auto translate_ip_list =
+        [&route_desc](const std::string& json_name,
+                      Protobuf::RepeatedPtrField<envoy::api::v2::CidrRange>* dest) {
+          if (route_desc->hasObject(json_name)) {
+            for (const std::string& source_ip : route_desc->getStringArray(json_name)) {
+              Network::Address::CidrRange cidr(Network::Address::CidrRange::create(source_ip));
+              if (!cidr.isValid()) {
+                throw EnvoyException(fmt::format("Invalid {} entry: {}", json_name, source_ip));
+              }
+              envoy::api::v2::CidrRange* v2_cidr = dest->Add();
+              v2_cidr->set_address_prefix(cidr.ip()->addressAsString());
+              v2_cidr->mutable_prefix_len()->set_value(cidr.length());
+            }
+          }
+        };
+
+    translate_ip_list("source_ip_list", route->mutable_source_ip_list());
```

Comment:
Can you factor this out into address_json.{h,cc}? I think this CIDR list util will be generally useful.

Reply:
done